src/components/pages/UserList/UserList.tsx

```diff
@@ -48,32 +78,68 @@ class UsersList extends Component<{}, State> {
           error: false,
         });
       })
-      .catch((err) => {
+      .catch(() => {
         this.setState({ loading: false, error: true });
       });
-  }
+  };
 
   handlePageClick = (data: { selected: number }) => {
-    this.setState({ currentPage: data.selected }, () => {
-      this.fetchUsers();
-    });
+    this.setState({ currentPage: data.selected }, this.fetchUsers);
+  };
+
+  handleModalOpen = (type: ActionType, user: User | null = null) => {
+    this.setState({ modalState: { type, user } });
   };
 
-  handleUserAdd = () => {
+  handleModalClose = () => {
+    this.setState({ modalState: { type: null, user: null } });
+  };
+
+  handleUserAddOrEdit = () => {
     this.setState({
       currentPage: 0,
     });
     this.fetchUsers();
+    this.handleModalClose();
+  };
+
+  toggleActionModal = (user: User | null) => {
+    this.setState({ selectedUser: user });
   };
 
   render() {
-    const { users, loading, limit, totalUsers, currentPage, error } =
-      this.state;
+    const {
+      users,
+      loading,
+      limit,
+      totalUsers,
+      currentPage,
+      error,
+      message,
+      messageType,
+      modalState,
+    } = this.state;
     const pageCount = Math.ceil(totalUsers / limit);
 
     return (
-      <div className="user-list-container">
-        <Header handleUserAdd={this.handleUserAdd} />
+      <div className="user-list-container" ref={this.box}>
+        <Header
+          toggleAddUserModal={() => this.handleModalOpen(ActionType.add)}
+        />
+
+        {message.length > 0 && (
+          <Notification message={message} type={messageType} />
+        )}
+        {(modalState.type === ActionType.add ||
```

Comment:
```suggestion
        {[ActionType.add, ActionType.edit].includes(modalState.type) &&
```
use `includes` method